src/OrbitGl/ThreadTrack.cpp

```diff
@@ -501,8 +506,19 @@ void ThreadTrack::UpdatePrimitives(Batcher* batcher, uint64_t min_tick, uint64_t
         batcher->AddVerticalLine(pos, box_height, draw_data.z, color, std::move(user_data));
       }
 
-      // Use the time at boundary of the next pixel as a threshold to avoid overdraw.
-      next_pixel_start_time_ns = GetNextPixelBoundaryTimeNs(pos[0] + size[0], draw_data);
+      // Optimization when going through the scopes, as it doesn't have any sense to draw two timers
+      // in the same pixel.
+      // TODO (b/199736246): NextPixelBoundaryTimeNs should be greater than the pixel provided.
+      next_pixel_start_time_ns =
+          std::max(timer_info.end() + 1, GetNextPixelBoundaryTimeNs(pos[0] + size[0], draw_data));
```

Comment:
That function is sometimes giving us a previous timestamp (explained in detail in b/199736246) and I looked there for an hour because in that case we have an infinite loop. I will address it, but I can do it before in another PR if it's cleaner.

Reply:
I'd maybe add that logic to GetNextPixelBoundaryTimeNs itself by passing in timer_info.end() as argument so that it's clearer that the problem lies in the function.

Reply:
Good idea

Reply:
I prefer to fix the issue of `GetNextPixelBoundaryTimeNs` function first. Adding `timer_info.end() + 1` feels like a hack.

Reply:
Ok, I will first that function first.

Reply:
All this logic has been replaced by a function inside ScopeTree